tensorflow/python/ops/image_ops_impl.py

```diff
@@ -3541,3 +3542,63 @@ def combined_non_max_suppression(boxes,
     return gen_image_ops.combined_non_max_suppression(
         boxes, scores, max_output_size_per_class, max_total_size, iou_threshold,
         score_threshold, pad_per_class)
+  
+  
+@tf_export('image.median_filtering_2D')
+def median_filtering_2D(tf_img,filter_shapex=3,filter_shapey=3):
+    # This methods takes both 2D Tensor as well as 3D Tensor Images
+    # Other than Tensor it takes optional parameter filter_Size
+    # Default Filter Size = 3 , 3
+    # This Median Filtering is done by using 2D filters of user's choice
+    # Filter_size should be odd
+    # This method takes both kind of images where pixel values lie between 0 to 255 and where it lies between 0.0 and 1.0
+    
+    try:
+        m,no = int(tf_img.shape[0]),int(tf_img.shape[1])
+    except:
```

Comment:
A naked try-except is really bad as it can mask errors unforeseen by the original author, specially as code evolves.

The message you suggest below, for example, does not match reality. The input tensor can be an image of partially unknown shape and the exception will fire; similarly it can be not an image and the exception will not fire.